i3_agenda/event_utils.py

```diff
@@ -0,0 +1,49 @@
+
+import time
+import datetime
+from typing import Optional, List
+
+from event import Event
+
+
+def get_closest(events: List[Event], hide_event_after: int) -> Optional[Event]:
+    closest = None
+    for event in events:
+        # Don't show all day events
+        if event.is_allday:
+            continue
+
+        now = time.time()
+        # If the event already ended
+        if event.end_time < now:
+            continue
+
+        # If the event started more than hide_event_after ago
+        if hide_event_after > -1 and event.unix_time + 60 * hide_event_after < now:
+            continue
+
+        if closest is None or event.unix_time < closest.unix_time:
+            closest = event
+
+    return closest
+
+
+def get_event_time(full_time: str) -> float:
+    if "T" in full_time:
+        event_time_format = "%Y-%m-%dT%H:%M:%S%z"
+    else:
+        event_time_format = "%Y-%m-%d"
+
+    # Python introduced the ability to parse ":" in the timezone format (in strptime()) only from version 3.7 and up.
+    # We need to remove the : before the timezone to support older versions
+    # See https://stackoverflow.com/questions/30999230/how-to-parse-timezone-with-colon for more information
+    if full_time[-3] == ":":
+        full_time = full_time[:-3] + full_time[-2:]
+
+    return time.mktime(
+        datetime.datetime.strptime(full_time, event_time_format).astimezone().timetuple()
+    )
+
+
+def is_allday(start_time: str) -> bool:
```

Comment:
this function could be a method of the `Event` class

Reply:
Now it is a method of Event, but it infer if it is all day, looking at the unix_time and checking if it begins at midnight, it can be questionable. So if you see another approach to that please propose me.

Reply:
This isn't a good solution, events could start at 00.00.00 (for example a party or a date or whatever) and not be an all day event, the static constructor mentioned above solves this issue for us, since we could have the JSON in the "constructor" (not really the constructor, but the static builder), we could just check it like we did in the old way